core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1464,9 +1464,12 @@ class Log(@volatile var dir: File,
       var segmentEntry = segments.floorEntry(startOffset)
 
       // return error on attempt to read beyond the log end offset or read below log start offset
-      if (startOffset > endOffset || segmentEntry == null || startOffset < logStartOffset)
-        throw new OffsetOutOfRangeException(s"Received request for offset $startOffset for partition $topicPartition, " +
-          s"but we only have log segments in the range $logStartOffset to $endOffset.")
+      if (startOffset > endOffset || segmentEntry == null || startOffset < logStartOffset) {
+        val msg = s"Received request for offset $startOffset for partition $topicPartition, " +
+          s"but we only have log segments in the range $logStartOffset to $lastStableOffset."
```

Comment:
The use of `lastStableOffset` is incorrect. The check above uses `endOffset`, so we should probably use that in the message as well.

Reply:
Refactored